lib/queue.js

```diff
@@ -615,17 +637,14 @@ Queue.prototype.processJob = function(job){
   //
   var renew = false;
   var lockRenewer = function(){
-    // The first call to lock the job should ensure that the job is in the 'active' state,
-    // because it might have gotten picked up already by another processor. We don't need
-    // to do this on subsequent calls.
-    return job.takeLock(_this.token, renew, !renew).then(function(locked){
-      if(locked){
+    return job.takeLock(renew, true).then(function(lock){
```

Comment:
Here [`takeLock` returns early](https://github.com/nextorigin/bull/blob/5c1e9d9b5e8e480050b558c197e37a48210b924c/lib/scripts.js#L327) and relies on redlock to extend the lock, `ensureActive` is only considered for new locks.  So I don't know if insinuating the second parameter matters on a renew is necessary?